vars/kibanaCoverage.groovy

```diff
@@ -0,0 +1,193 @@
+        "$(${CMD}":%an" -1)" \
+        "$(${CMD}":%s" -1)")
+    touch VCS_INFO.txt
+    for X in "${!XS[@]}"; do
+    {
+      predicate "${XS[X]}"
+      echo "${XS[X]}" >> VCS_INFO.txt
+    }
+    done
+    echo "### VCS_INFO:"
+    cat VCS_INFO.txt
+    ''', title
+  )
+}
+
+def bootMergeAndIngest(buildNum, buildUrl, title) {
+  kibanaPipeline.bash("""
+    source src/dev/ci_setup/setup_env.sh
+    # bootstrap from x-pack folder
+    cd x-pack
+    yarn kbn bootstrap --prefer-offline
+    # Return to project root
+    cd ..
+    . src/dev/code_coverage/shell_scripts/extract_archives.sh
+    . src/dev/code_coverage/shell_scripts/fix_html_reports_parallel.sh
+    . src/dev/code_coverage/shell_scripts/merge_jest_and_functional.sh
+    . src/dev/code_coverage/shell_scripts/copy_mocha_reports.sh
+    . src/dev/code_coverage/shell_scripts/ingest_coverage.sh ${buildNum} ${buildUrl}
+  """, title)
+}
+
+def ingestWithVault(buildNum, buildUrl, title) {
+  def vaultSecret = 'secret/kibana-issues/prod/coverage/elasticsearch'
+  withVaultSecret(secret: vaultSecret, secret_field: 'host', variable_name: 'HOST_FROM_VAULT') {
+    withVaultSecret(secret: vaultSecret, secret_field: 'username', variable_name: 'USER_FROM_VAULT') {
+      withVaultSecret(secret: vaultSecret, secret_field: 'password', variable_name: 'PASS_FROM_VAULT') {
+        bootMergeAndIngest(buildNum, buildUrl, title)
+      }
+    }
+  }
+}
+
+def ingest(timestamp, title) {
+  withEnv([
+    "TIME_STAMP=${timestamp}",
+  ]) {
+    ingestWithVault(BUILD_NUMBER, BUILD_URL, title)
+  }
+}
+
+def runTests() {
+  parallel([
+    'kibana-intake-agent': workers.intake('kibana-intake', './test/scripts/jenkins_unit.sh'),
+    'x-pack-intake-agent': {
+      withEnv([
+        'NODE_ENV=test' // Needed for jest tests only
+      ]) {
+        workers.intake('x-pack-intake', './test/scripts/jenkins_xpack.sh')()
+      }
+    },
+    'kibana-oss-agent'   : workers.functional(
+      'kibana-oss-tests',
+      { kibanaPipeline.buildOss() },
+      ossProks()
+    ),
+    'kibana-xpack-agent' : workers.functional(
+      'kibana-xpack-tests',
+      { kibanaPipeline.buildXpack() },
+      xpackProks()
+    ),
+  ])
+}
+
+def ossProks() {
+  return [
+    'oss-ciGroup1' : kibanaPipeline.ossCiGroupProcess(1),
+    'oss-ciGroup2' : kibanaPipeline.ossCiGroupProcess(2),
+    'oss-ciGroup3' : kibanaPipeline.ossCiGroupProcess(3),
+    'oss-ciGroup4' : kibanaPipeline.ossCiGroupProcess(4),
+    'oss-ciGroup5' : kibanaPipeline.ossCiGroupProcess(5),
+    'oss-ciGroup6' : kibanaPipeline.ossCiGroupProcess(6),
+    'oss-ciGroup7' : kibanaPipeline.ossCiGroupProcess(7),
+    'oss-ciGroup8' : kibanaPipeline.ossCiGroupProcess(8),
+    'oss-ciGroup9' : kibanaPipeline.ossCiGroupProcess(9),
+    'oss-ciGroup10': kibanaPipeline.ossCiGroupProcess(10),
+    'oss-ciGroup11': kibanaPipeline.ossCiGroupProcess(11),
+    'oss-ciGroup12': kibanaPipeline.ossCiGroupProcess(12),
+  ]
+}
+
+def xpackProks() {
```

Comment:
thoughts on pulling the list of ci groups from jobs.yml?  we haven't changed groups in a while so this is in the nice to have category, but they have the potential to be fluid

Reply:
I didn't write this part.  Perhaps @dmlemeshko can help here?

Reply:
@jbudz we don't run all the ci group listed in jobs.yml, so it is still a bit manual work to select the required ones.
We already have [Jenkinsfile_coverage](https://github.com/elastic/kibana/blob/master/.ci/Jenkinsfile_coverage#L20) in master listing groups the same way they are in the main [Jenkinsfile](https://github.com/elastic/kibana/blob/master/Jenkinsfile#L14)  Maybe for simplicity we can just keep those 2 files similar in part of ci groups?

Reply:
cool, makes sense.  we can always refactor another time if turns into an issue